### ISSUE TITLE
Service Worker breaks Universal SSR

### ISSUE THREAD

**Issue by dottodot** [angular/angular#20890]
## I'm submitting a...

<pre><code>
[ ] Regression (a behavior that used to work and stopped working in a new release)
[x] Bug report  
[ ] Feature request
[ ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
</code></pre>

I'm sure my service worker was working with SSR at one point but for some reason it's now preventing SSR working.

I'm importing it into my app.browser.module like

```
  environment.production
      ? ServiceWorkerModule.register('/ngsw-worker.js')
      : []
```
I'm not getting any errors on client or server, all I know is if I remove the service worker SSR works again.

I've also tried it just in my main app.module as I believe it's meant to be Universal friendly now but I get the same result.

It seems odd that there is no error as usually there is one when something is preventing the ssr.

## Environment
<pre><code>
Angular CLI: 1.6.0
Node: 8.9.1
OS: darwin x64
Angular: 5.1.0
... animations, common, compiler, compiler-cli, core, forms
... http, language-service, platform-browser
... platform-browser-dynamic, platform-server, router
... service-worker

@angular/cli: 1.6.0
@angular-devkit/build-optimizer: 0.0.35
@angular-devkit/core: 0.0.22
@angular-devkit/schematics: 0.0.41
@ngtools/json-schema: 1.1.0
@ngtools/webpack: 1.9.0
@schematics/angular: 0.1.10
@schematics/schematics: 0.0.10
typescript: 2.4.2
webpack-bundle-analyzer: 2.9.1
webpack: 3.10.0
</code></pre>

**Comment by dottodot** [comment]
I tried to replicate in a fresh install but can't so must be something particular to my app. Trouble is with no error to lead me in the right direction I'm not sure how I can find the cause. 

What I have found though is when the is no service worker and I view page source I get the result I expect when ssr is working. With the service worker in place view page source look like ssr is not working but if I refresh the view source page it then looks like ssr is working.

**Comment by odahcam** [comment]
Open the page in Chrome and type Ctrl + U. If SSR is working, you should see rendered by the server HTML, otherwise there will be only Angular tags.

**Comment by dottodot** [comment]
OK well that is the same as view source. 

So when the page loads I just see angular tags but if I clear site data i.e cache storage, application cache I can see the server rendered html. If I refresh it returns to only getting angular tags.

**Comment by odahcam** [comment]
I think thats the desired behavior 'cause as far as i know, that's what happens:

1. First access: server renders and deliver the application to the client (pre-rendered Angular app).
2. Client caches/stores application's data.
3. Client uses the Angular app normally.
4. Client leaves.
5. Client gets back, service-worker gives the client all the cached/stored data (fresh Angular app).
6. Client uses the Angular app normally.

That's it! I think it should work that way.

**Comment by diegotham** [comment]
It happens to me as well. The only thing I do different from the docs is adding an interceptor as explained in here: https://github.com/cyrilletuzi/nguniversal-expressengine-httpinterceptor

If SSR works, service worker does not, and viceversa

**Comment by dottodot** [comment]
@odahcam I'm not entirely convinced that's correct. If the first thing is the server returning the pre rendered app I should always be able to see the rendered html. Furthermore I've had the 2 working together in the past. 

The only time I've ever seen unrendered html is when there is some error or issue that is preventing it from working.

I'm happy to be proved wrong but after always seeing server rendered html it's a bit disconcerting to now not see it.

**Comment by alxhub** [comment]
Fixed as of 5.1.0!

**Comment by dottodot** [comment]
@alxhub please can you explain why you've closed this issue. I reported the as being in 5.1.0 so can't see how it's fixed as of 5.1.0
